### ISSUE TITLE
snapshot: File name of screenshots is appended by a unique ID

### ISSUE THREAD

**Issue by funnel20** [fastlane/fastlane#8339]
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I have read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)

### Issue Description
After a successful run of Fastlane SnapShot, the files are copied to the defined `output_directory` in my SnapFile. However, the *.png files all have some kind of ID in the name, e.g.:
_iPhone5s-01_SmartTab-589996837d813aace56d9a0cf7b764a0.png_

This is the Swift code in the Test file:
`snapshot("01_SmartTab")`

When I ran SnapShot in the past, the expected file name was generated:
_iPhone5s-01_SmartTab.png_

The HTML report still uses the correct file names, so my original screenshots are displayed, instead of the new image files.

This might have to do with the option `clear_previous_screenshots` which is set to `false`.
This is done deliberately, since our screenshots are managed under Subversion.
When the option is set to `true` the SVN working copy will be destroyed when the files are deleted.

##### Complete output when running fastlane, including the stack trace and command used
> You can use: `--capture_output` as the last commandline argument to get that collected for you

```
[17:56:02]: ▸ ✓ testCreateSnapShots (55.287 seconds)
[17:56:02]: ▸ 	 Executed 1 test, with 0 failures (0 unexpected) in 55.287 (55.292) seconds
[17:56:02]: ▸ 
[17:56:02]: Collecting screenshots...
[17:56:02]: Found 2 screenshots...
[17:56:02]: Copying '/Users/xxx/screenshots/en-US/iPhone5s-01_SmartTab-589996837d813aace56d9a0cf7b764a0.png'...
[17:56:02]: Copying '/Users/xxx/screenshots/en-US/iPhone5s-02_RemoteTab-589996837d813aace56d9a0cf7b764a0.png'...

+-----------+-------+
| snapshot results  |
+-----------+-------+
| Device    | en-US |
+-----------+-------+
| iPhone 5s |  💚   |
+-----------+-------+
```

### Environment

Please run `fastlane env` and copy the output below. This will help us help you :+1:
If you used `--capture_output` option please remove this block - as it is already included there.

<details><summary>🚫 fastlane environment 🚫</summary>

### Stack

| Key                 | Value                                       |
| ------------------- | ------------------------------------------- |
| OS                  | 10.12.3                                     |
| Ruby                | 2.2.4                                       |
| Bundler?            | false                                       |
| Git                 | git version 2.10.1 (Apple Git-78)           |
| Installation Source | ~/.rvm/rubies/ruby-2.2.4/bin/fastlane       |
| Host                | Mac OS X 10.12.3 (16D32)                    |
| Ruby Lib Dir        | ~/.rvm/rubies/ruby-2.2.4/lib                |
| OpenSSL Version     | OpenSSL 1.0.2j  26 Sep 2016                 |
| Is contained        | false                                       |
| Is homebrew         | false                                       |
| Xcode Path          | /Applications/Xcode.app/Contents/Developer/ |
| Xcode Version       | 8.2.1                                       |

### System Locale

| Error                       |
| --------------------------- |
| No Locale with UTF8 found 🚫 |

### fastlane files:

**No Fastfile found**

**No Appfile found**

### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.18.2  | ✅ Up-To-Date  |

### Loaded fastlane plugins:

**No plugins Loaded**

<details><summary><b>Loaded gems</b></summary>

| Gem                       | Version      |
| ------------------------- | ------------ |
| executable-hooks          | 1.3.2        |
| bundler-unload            | 1.0.2        |
| rubygems-bundler          | 1.4.4        |
| bundler                   | 1.13.6       |
| slack-notifier            | 1.5.1        |
| rouge                     | 1.11.1       |
| xcpretty                  | 0.2.4        |
| terminal-notifier         | 1.7.1        |
| unicode-display_width     | 1.1.1        |
| terminal-table            | 1.7.3        |
| plist                     | 3.2.0        |
| addressable               | 2.3.8        |
| multipart-post            | 2.0.0        |
| word_wrap                 | 1.0.0        |
| tty-screen                | 0.5.0        |
| babosa                    | 1.0.2        |
| colored                   | 1.2          |
| highline                  | 1.7.8        |
| commander                 | 4.4.0        |
| excon                     | 0.54.0       |
| faraday                   | 0.10.0       |
| unf_ext                   | 0.0.7.2      |
| unf                       | 0.1.4        |
| domain_name               | 0.5.20161021 |
| http-cookie               | 1.0.3        |
| faraday-cookie_jar        | 0.0.6        |
| fastimage                 | 1.9.0        |
| gh_inspector              | 1.0.2        |
| uber                      | 0.0.15       |
| representable             | 2.3.0        |
| retriable                 | 2.1.0        |
| mime-types-data           | 3.2016.0521  |
| mime-types                | 3.1          |
| hurley                    | 0.2          |
| little-plugger            | 1.1.4        |
| multi_json                | 1.12.1       |
| logging                   | 2.1.0        |
| jwt                       | 1.5.6        |
| memoist                   | 0.15.0       |
| os                        | 0.9.6        |
| signet                    | 0.7.3        |
| googleauth                | 0.5.1        |
| httpclient                | 2.8.3        |
| google-api-client         | 0.9.20       |
| json                      | 1.8.1        |
| mini_magick               | 4.5.1        |
| multi_xml                 | 0.5.5        |
| rubyzip                   | 1.1.7        |
| security                  | 0.1.3        |
| xcpretty-travis-formatter | 0.0.4        |
| dotenv                    | 2.1.1        |
| faraday_middleware        | 0.10.1       |
| i18n                      | 0.7.0        |
| thread_safe               | 0.3.5        |
| tzinfo                    | 1.2.2        |
| activesupport             | 4.2.7.1      |
| CFPropertyList            | 2.3.4        |
| claide                    | 1.0.1        |
| nanaimo                   | 0.2.3        |
| xcodeproj                 | 1.4.1        |
</details>

*generated on:* **2017-02-22**
</details>

**Comment by mfurtak** [comment]
@hjanuschka Can you comment on this?

**Comment by hjanuschka** [comment]
@funnel20 - the reason the "random" id added is that you can run snapshot with different launch arguments, and if the name is not uniq it would result in overwritten screenshots.

i tested it right now and it looks like that it works as intended. -> https://gist.github.com/hjanuschka/7f65946e1eabb2b3f4983586e68b392c

the screenshots have the uniq id, and the resulting html also has them.


i think you will need to, either change the html filxs by hand 😢  (the "uniq" has stays the same unless you are changing the launch_arguments) or re run snapshot once (to get the html).


let me know if this helps

**Comment by funnel20** [comment]
@hjanuschka - My problem is not that the resulting HTML is incorrect. 
My problem is that I don't want to have that unique ID in the file name of the created screenshot files.

Why is it there? Or more important, how do I get it rid of it?
If I define this name in my test file:
`snapshot("01_SmartTab")`

I expect that for each device type the *.png file name exactly reflects this name, and not with some weird random addendum. Conform the instructions at https://github.com/fastlane/fastlane/tree/master/snapshot.
This was the case in the past (around November 2016 I ran SnapShot for the last time).
Now with version 2.18.2 I get this addendum.

This is the only launch argument that is in my SnapFile:
`launch_arguments(["SnapShotTestActive"])`

Any help is appreciated.

**Comment by hjanuschka** [comment]
@funnel20  right now there is no option to disable the appending of uniq-hash.
and you are right documenation can be improved, and the readme didn't get updated.


this is the PR that introduced the change: https://github.com/fastlane/fastlane/pull/7670


i think i still didn't get it 100%, unless you change the `launch_arguments` - the uniq-hash stays the same - so you could always relay on it and use it in the html or wherever else.

**Comment by funnel20** [comment]
@hjanuschka - This technical solution for #7670 is an undesired solution from the perspective of usability.
Why add a MD5 hash to a filename? What if Windows Explorer or Mac Finder would do this automatically for each file that you save?
As I see, the changed code will always create the same filename when the launch arguments stay the same. 
Why is the MD5 required? I don't see a difference between always using a filename `01_SmartTab` or `01_SmartTab-<MD5 hash>`?

Apparently I have to write my own post-SnapShot file to strip the MD5 section from the file names.
I think this is contradicting the productivity benefits of FastLane.

**Comment by hjanuschka** [comment]
@funnel20  i am sorry for you, but i would love to review a PR that bypasses the hashed filenames unless launch_arguments are set, so if you are up to a PR just direct ping me.

**Comment by funnel20** [comment]
@hjanuschka - Is the following requirement an acceptable solution? I think this helps anyone.

Users who want to use launch arguments should have an option to omit the MD5 hash in the screenshot filenames. This can be achieved by adding the pre-defined launch argument: `omitHashInOutputFilename` (or something more to your liking).

When this launch argument is available, the code will output the original filenames as requested.
So for users who are fine with the current implementation don't have to change anything, users like me can opt-in to omit the hash.

**Comment by hjanuschka** [comment]
i think that it would be easier to go with something like this:


  * do not append uniq id, if launcharguments is not sit (== nil)
  * do not append uniq id, if launch arguments array is exactly the length of 1 

this way it should be only appended in situations where launch arguments are heavily used.
what do you think about it?

if you want/can provide a PR, i am happily helping you if you need any assistance.

**Comment by funnel20** [comment]
@hjanuschka 
As a user this would solve my problem. But only temporary.
What if I decide to use 2 launch arguments in the (near) future?

From the perspective of a Software Engineer the solution is bad. 
The software behaviour is changed significantly based on a subjective count of an input argument.
Since this is a fundamental discussion about the SnapShot product, I hope that @KrauseFx can join this discussion.

My biggest objection is that the solution for a specific use-case (#7670) affects **all users** (who use `launch_arguments`).
By trying to understand the use-case, I hope to provide an appropriate solution.

**Use-case #7670**
A user wants to run Snapshot several times in a row, with different `launch_arguments`.
As the same file names for the screenshots are being used, each new Snapshot run will overwrite the screenshots of the previous Snapshot run.
In the end there is only 1 set of screenshots that match the `launch_arguments` of the last Snapshot run.

_I don't really understand the reason for this use-case._
In the end, one can upload only 1 set of screenshots to the App Store.
Maybe the reason is that the user wants to compare different sets of screenshots based on the launch arguments. And decide which set is best suitable for the App Store.
Anyway, let's analyse the current solution.

**Current solution #7670**
Whenever `launch_arguments` is used, the file name of the screenshots will be appended by a unique ID. This ID is the MD5 hash of the `launch_arguments`.
So when there are 3 Snapshot runs, the output folder consists of:
`<device name>-<snapshot name>-<ID 1>.png`
`<device name>-<snapshot name>-<ID 2>.png`
`<device name>-<snapshot name>-<ID 3>.png`

**Issues with current solution**

1. How to connect the different IDs with the `launch_arguments`?
Each set of `launch_arguments` represents a specific situation. E.g. "blue theme", "user John Appleseed", etc. It's inconvenient to match these situations with the fixed and randomly ID.
2. How do you upload the screenshots with Fastlane Deliver?
Can this handle the multiple file names for the same screenshot in the same folder?
Or should you delete all the screenshots (manually) which are not selected, resulting in a single file for each screenshot? This sounds tedious.
3. All users who are not interested in this particular use-case will have this undesired ID in their file names.
So for a feature requested by a minority of users, the majority is punished.

**Alternative solution with original Snapshot version**
The following solution with Snapshot prior to the fix of #7670 will solve all the 3 above issues and fulfils the use-case too. 

Use a different value for the parameter `output_directory` for each different set of `launch_arguments`.
So when there are 3 Snapshot runs, the output folder structure will be:
`<folder situation 1>/<device name>-<snapshot name>.png`
`<folder situation 2>/<device name>-<snapshot name>.png`
`<folder situation 3>/<device name>-<snapshot name>.png`

This solves all 3 of the above issues:

1. The user has influence of specifying `output_directory`. So the folder with screenshots is directly and understandably connected to the test situation.
2. After deciding on the best set of screenshots for the App Store, that folder content can be copied to the default `screenshots` folder for Deliver.
3. No user is bothered with the ID and gets his desired file name as specified in the test script via `snapshot()`.

In other words, the change of #7670 can be reverted, as this use-case is covered by using `output_directory`.

Please let me know if any of my assumptions are incorrect.

**Comment by lacostej** [comment]
👍  to specifying the output directory. Also we have to make sure the launch arguments don't make it into the file name, right?

**Comment by mfurtak** [comment]
The discussion here is really great. Thanks for working this out together! 💯

**Comment by funnel20** [comment]
@lacostej - If using the `output_directory` is the proper solution for #7670 as I describe in my previous comment, indeed the MD5 hash (based on content of launch_arguments) can be removed from the file name.
Basically this is reverting the fix for #7670, as indicated in https://github.com/fastlane/fastlane/pull/7670/files.

But first I want to gain confidence that my suggestion with `output_directory` is indeed an appropriate solution for the use-case.
Is there anyone who disagrees?

**Comment by hjanuschka** [comment]
@funnel20 i think this just moves the error/defect of broken filenames to foldernames. (however i like the idea of having folders per launch_argument)
as the PR that introduced the md5 hashes, tried to fix non-easy launch_arguments.

like: `launch_arguments: ["-myUrl https://google.com"]` - this resulted in bad/broken filenames.
a hash was the only future-proof solution i could come up with, but i would love to hear what a good alternative soltion would be.

**Comment by funnel20** [comment]
@hjanuschka 
Can you explain in depth "the error/defect of broken filenames"?
- What exactly is the use-case that resulted in the fix for #7670?
- What is the expected result?
- What was the actual resulted before the fix?

If you use 1 folder for per `launch_arguments`, I don't see any reason to append whatever string in the file names. Why should there be any relation between the content of `launch_arguments`?
What are "non-easy launch_arguments"? It's just a bunch of characters.

**Comment by hjanuschka** [comment]
@funnel20 

the initial issue where filenames resulting in:
`/Users/MyApp/fastlane/screenshots/en-US/iPhone4s--myUrl https://google.com-0Launch.png`

which later on failed to copy

```
[14:49:28]: Copying '/Users/MyApp/fastlane/screenshots/en-US/iPhone4s--myUrl https://google.com-0Launch.png'...
[14:49:28]: No such file or directory @ rb_sysopen - /Users/MyApp/fastlane/screenshots/en-US/iPhone4s--myUrl https://google.com-0Launch.png
```


> What exactly is the use-case that resulted in the fix for #7670?
i think user wants to run snapshot with different urls in the app

> What is the expected result?
Get screenshots of different settings, and for sure do *not crash*

> What was the actual resulted before the fix?
Crashed, and snapshot didn't finish successfully.



>What are "non-easy launch_arguments"?
e.g.: `-`. i know that its just a bunch of characters, but not all characters are good characters in context of file operations.

**Comment by funnel20** [comment]
I know see that #7670 was a fix for #7663.
This explains a lot.
The original code created the screenshot name as:
`<device name>-<launch arguments>-<snapshot name>.png`

I can see why this fails to copy, as the bunch of characters from `launch_arguments` might create an invalid file path.

Fix #7670 changes this to:
`<device name>-<snapshot name>-<MD5 hash>.png`

**Comment by hjanuschka** [comment]
@funnel20  exactly! and your concerns are right about unpredictable filenames.
but i think, not crashing has the higher priority in this case. but i am still open for improvements and changes let me know if you are going for a PR 👍

**Comment by funnel20** [comment]
@hjanuschka 
Combing #7663, your answer and good practise of software engineering that it will never crash:

**Use-case**
The user wants to run snapshot with different `launch_arguments` in the app.

**Expected result**
Get screenshots of different settings.

This matches with my assumptions from my earlier post 👍.
What doesn't match, is my solution to revert to the original code from before fix #7670.
Because that will introduce the `launch_arguments` into the file name and Snapshot will crash.

So my proposed solution will be:

1. Remove `launch arguments` from the file name. This will always create the file name as:
`<device name>-<snapshot name>.png`
_This was the situation in November 2016, so does anybody know why `launch arguments` was introduced in the file name?_ 
2. Use `output_directory` to have unique screenshot locations for the different runs.

**Comment by hjanuschka** [comment]
i still think my proposed solution

  * do not append uniq id, if launcharguments is not set (== nil)
  * do not append uniq id, if launch arguments array is exactly the length of 1

(i think a similar approach was the original behaviour introduced with: https://github.com/fastlane-old/snapshot/commit/02509add9a99a51b7f843298d2bd3ecbb42a2783)

is the one that should make all stakeholders happy, and not introduce too much new code.

**Comment by funnel20** [comment]
@hjanuschka 
I tried to convince you earlier that your solution will not make me happy.
Also the product and its documentation will become spaghetti.

1. I will have the same issue again when I use 2 or more `launch_arguments`.
2. It's bad practise to base software behaviour on a subjective count of an input argument. And creates unnecessary complicated documentation.
3. It's a workaround for an incorrect fix for a problem that wasn't a problem. I.e. it was unnecessary to put the `launch_arguments` into the file name (as `output_directory` can be used), which was than solved by creating a MD5 hash, which would now be removed for certain conditions in order to partly fix #8339.

I think we agreed that the use-case is:
**The user wants to run snapshot with different `launch_arguments` in the app.**

As this can be fulfilled with my solution of using a unique `output_directory` per test run, it does NOT require a unique ID
I don't introduce any new code. Instead, line 30 of collector.rb will be straightforward:
`screenshot_name = device_name + "-" + name + ".png"`

instead of:
`screenshot_name = device_name + "-" + name + "-" + Digest::MD5.hexdigest(components.join("-")) + ".png"`

Also it doesn't require any update to the documentation. In fact the software will behave exactly as the current documentation, as the docu isn't currently reflecting the MD5.

I didn't hear any argument why my proposal isn't correct. 
Why wouldn't it be preferred over yours?

**Comment by hjanuschka** [comment]
`launch_arguments` and it is iterated over the amount of devices.

so extending your sample:

```
lane :sd do
	snapshot(project: "snapshot/example/Example.xcodeproj",
          scheme: "Example", 
          output_directory: "google",
          launch_arguments: ["-myUrl https://google.com", "-myUrl https://alphabet.com"], 
          devices: ["iPhone 5s"]
        )
end
```


would end up putting alphabet screenshots into the "google" folder. where as it is expected to produce seperated files.

**Comment by funnel20** [comment]
@hjanuschka 
Finally I get it. Sorry for being so unbelievable stupid. 
It would have helped if it was brought up that it's possible to **repeat the same key** with multiple values in the `launch_arguments`.

Based on your reply with 2 times key `myUrl` in `launch_arguments`, I read the documentation again.
In the last part of section "Launch Arguments" is mentioned that a single Snapshot run will be repeated automatically for A/B testing when the same key is repeated in `launch_arguments`:

```PHP
# Snapfile for A/B Test Comparison
launch_arguments([
  "-secretFeatureEnabled YES",
  "-secretFeatureEnabled NO"
])
```

Although my proposal might still be used by starting Snapshot twice with a different `output_directory`:
```ruby
 lane :sd do
	snapshot(project: "snapshot/example/Example.xcodeproj",
          scheme: "Example", 
          output_directory: "google",
          launch_arguments: ["-myUrl https://google.com"], 
          devices: ["iPhone 5s"]
        )
end

lane :sd do
	snapshot(project: "snapshot/example/Example.xcodeproj",
          scheme: "Example", 
          output_directory: "alphabet",
          launch_arguments: ["-myUrl https://alphabet.com"], 
          devices: ["iPhone 5s"]
        )
end
```
Although I understand that comparing the output files for A/B testing is more easy hen the files are in the same folder.

In my understanding `launch_arguments` consisted of a set of **unique keys** which describe a single test case. Like:
```PHP
launch_arguments([
  "-firstName Felix",
  "-lastName Krause"
])
```

Hence that I was linking 1 set of `launch_arguments` to 1 Snapshot run and thus to 1 `output_directory`.

Back to the use-cases to discover a new solution...

There are 3 use-cases:

1. Snapshot is ran without `launch_arguments`. 
This will create a single screenshot. There is no need to make the file name unique.
2. Snapshot is run with `launch_arguments` which contains unique key(s).
The unique keys create a single test situation, hence this will output a single file for each screenshot. There is no need to make the file name unique.
3. Snapshot is run with `launch_arguments` which contains non-unique key(s).
The purpose is to facilitate A/B testing, where Snapshot repeats the test automatically for each value of the same. This will output multiple files for the same screenshot, hence the file name should be unique.

All 3 are technically possible with Snapshot.
Only use-case 2 fails, because the current implementation doesn't discern between use-case 2 and 3; whenever `launch_arguments` is present, the output files have the unique ID (this issue #8339 which I brought up).

**Solution**
The solution is indeed close to your proposal. Except that not the number of `launch_arguments` is the condition, but the uniqueness of it's keys.

- do not append uniq id, if `launch_arguments` is not set (== nil)
- do not append uniq id, if `launch_arguments` array ~~is exactly the length of 1~~ contains unique keys
- do append uniq id, if `launch_arguments` array contains at least 1 duplicate key
This detection is already in the current code, as it triggers a new test run.

Do you agree?

**Comment by lacostej** [comment]
Can someone explain how people use of multiple launch arguments? I am not exactly sure I understand the A/B testing use case. In the meantime, here's a proposal:

1. 0 or 1 launch arguments: do not append launch arguments to the screenshot name
2. 2 or more launch arguments, either sanitize it, or replace it with an index (e.g. `device_name + "-" + name +"-" + idx + ".png"`). Alternatively we could also pass a suffix array argument (`device_name + "-" + name +"-" + suffix[idx] + ".png"`

if someone runs it in multiple runs

**Comment by funnel20** [comment]
@lacostej 
Please read my 3 different use-case.
Your solution fails (similar to that of @hjanuschka ) for use-case 2, by defining the logic based on the number of launch arguments.

The existing Snapshot logic is that the test:

- runs once when the keys in `launch_arguments` are unique
So even when there are 6 unique keys, a single run is performed and there is no need to make the file names unique.
- runs multiple times if there are duplicate keys in `launch_arguments`.
Now the file names should be made unique.

**Comment by funnel20** [comment]
After some testing it appears that I interpreted `launch_arguments` wrong. Again.
Because it has a double function:

1. Each element in the array represents a new test run.
2. Multiple key/value pairs can be combined in 1 element to define the launch arguments for that particular test run.

To understand these, consider the next 2 examples:

**Example 1**
```ruby
launch_arguments([
  "-firstName Felix -lastName Krause"
])
```
The array has 1 element, with 2 _combined_ keys: `firstName` and `lastName`.
This will run Snapshot once, with both values set.

**Example 2**
```ruby
launch_arguments([
  "-firstName Felix",
  "-lastName Krause"
])
```
The array has 2 element, with each 1 key.
This will run Snapshot twice; one run with `firstName` and the second run with `lastName`.
As a result, this will output 2 versions of each screenshot.

Now I finally understand why @hjanuschka was using the count of 1 element as condition for adding the ID in the file name:

- do not append uniq id, if launcharguments is not set (== nil)
- do not append uniq id, if launch arguments array is exactly the length of 1

I would really recommend to update the documentation about `launch_arguments` at: https://github.com/fastlane/fastlane/tree/master/snapshot#launch-arguments

The main reason why I was so confused about the behaviour of this parameter is that the name doesn't entirely reflect it's purpose.
An array with name "launch arguments" implies that each element in the array is a single launch argument.
But adding elements to this array actually means creating new test runs.

**Comment by lacostej** [comment]
@funnel20 so given your last understanding, would my solution work? The uniq id isn't very descriptive. An index, a suffix, or a sanitized "launch arguments" array element would be better IMO.

**Comment by funnel20** [comment]
@lacostej 
This issue is about removing the undesired unique ID in the file name when the use-case doesn't require a unique file name.
So for situations where `launch_arguments` is not used, empty or has 1 element.

I like your idea of replacing the unique ID with a more descriptive string when `launch_arguments` contains more than 1 element.
But that's a different discussion, so feel free to open a new issue for this.

**Comment by funnel20** [comment]
@hjanuschka 
Should I make a PR for you?

**Comment by bonyadmitr** [comment]
Hi!
It is very important to disable unique ID!
We can't automate snapshot with frameit strings!
We need something like in example [MyDayToDo](https://github.com/fastlane/examples/blob/master/MyDayToDo/en-US/keyword.strings):

```
"1_iphone6_1" = "Day Tasks";
"2_iphone6_2" = "Share";
"3_iphone6_3" = "Speech";
"4_iphone6_4" = "Widget";
"5_iphone6_5" = "Reminders";
```

**Comment by bonyadmitr** [comment]
It will be good feature to disable unique ID!

**Comment by funnel20** [comment]
Multiple users want to have this ID removed.

**Comment by funnel20** [comment]
There was activity 9 days ago, what kind of stupid bot are you?
But as nobody seemed to have any intention of fixing this, what's the difference?

**Comment by SemenovAlexander** [comment]
Hey, I'm also facing the same issue and need to strip this unique id from filenames (snapshot now adds `uniqueId`, even if I pass no launch arguments at all, and it's very confusing.).

Actually, for now, I have to strip that id, because I rely on filenames for other tasks (screenshots framing via Sketch plugin).

**Comment by KrauseFx** [comment]
Hey everyone, yeah, that totally makes sense, and I'd be in favor of having an option to remove it. Feel free to submit a PR adding the new option to _snapshot_, check out our [Contributing.md](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md) on more information on how to get started contributing.

@funnel20 Please be respectful to others, no matter if human beings or bots. We're all trying our best here, and everyone can submit pull requests to this project 👍 Thanks ❤️

**Comment by carlosalban** [comment]
Would also love to have this issue fixed! Any update?
